src/phoenix/experimental/evals/templates/template.py

```diff
@@ -45,49 +52,45 @@ def _get_delimiters(self, delimiters: List[str]) -> Tuple[str, str]:
         else:
             raise ValueError("delimiters must only contain 2 items in the list")
 
-    def _validate(self) -> None:
+    def _validate(self, text: str) -> None:
         # Validate that for every open delimiter, we have the corresponding closing one
-        start_count = self.text.count(self._start_delim)
-        end_count = self.text.count(self._end_delim)
+        start_count = text.count(self._start_delim)
+        end_count = text.count(self._end_delim)
         if start_count != end_count:
             raise ValueError(
                 f"text poorly formatted. Found {start_count} instances of delimiter "
                 f"{self._start_delim} and {end_count} instances of {self._end_delim}. "
                 "They must be equal to be correctly paired."
             )
 
-    def _parse_variables(self) -> List[str]:
-        variables = []
-        formatter = Formatter()
-
-        text = self.text
-
-        # Example of this could be a template like: My name is ::name::
-        if self._start_delim == self._end_delim:
-            delim_length = len(self._start_delim)
-            delim_count = text.count(self._start_delim)
-            while delim_count > 0:
-                left_index = text.find(self._start_delim)
-                right_index = text[left_index + delim_length :].find(self._start_delim)
-                text = (
-                    text[0:left_index]
-                    + DEFAULT_START_DELIM
-                    + text[left_index + delim_length : left_index + delim_length + right_index]
-                    + DEFAULT_END_DELIM
-                    + text[left_index + 2 * delim_length + right_index :]
-                )
-                delim_count = text.count(self._start_delim)
-        else:
-            if self._start_delim != "{":
-                text = text.replace(self._start_delim, DEFAULT_START_DELIM)
-            if self._end_delim != "{":
-                text = text.replace(self._end_delim, DEFAULT_END_DELIM)
+    def _parse_variables(self, text: str) -> List[str]:
+        pattern = re.escape(self._start_delim) + "(.*?)" + re.escape(self._end_delim)
+        variables = re.findall(pattern, text)
+        return variables
 
-        for _, variable_name, _, _ in formatter.parse(text):
-            if variable_name:
-                variables.append(variable_name)
 
-        return variables
+class ClassificationTemplate(PromptTemplate):
```

Comment:
The sub-classing isn't really a necessary feature, it's just the mechanism I used for the first step of this refactor to keep the interfaces between the `ClassificationTemplate` and `PromptTemplate` the same (so they can be used interchangeably). One way this can be taken is to explicitly extract an abstract class to enforce this interface. Another is to let `llm_classify` handle each kind of input with a different code path.

My preference is to "tell don't ask", and not have `llm_classify` do any kind of type checking to know how to generate responses. By equipping the "raw user string" style templates and `phoenix`-provided templates with the same interface we can keep `llm_classify` from needing to know about all of the different implementations

Reply:
I don't think templates and configuration/ tasks are the same thing. For example, this feels like we wind up putting too much functionality into the `format` method on the classification template, which is now responsible for both selecting a template and applying variables. It feels much more natural to me to have a classification task or configuration that does not share an interface with the template.

Reply:
I definitely think it's possible there might be an underlying abstraction here but I'm not sure we should push for too much refactoring just yet. If users can pass in objects with two separate interfaces (tasks vs templates) then we need to handle two completely different types of objects that might need to support parallel functionality. While I'm not sure that `Templates` are the right idea here, I do think `llm_classify` should work against a single standard interface

Reply:
I think this is a fair concern. Let me work it out with @anticorrelator